### PR TITLE
Clean up guidance docs

### PR DESCRIPTION
**Pull request by reyang**
* Fixed some caveats such as broken links, grammar issues, inconsistency, etc.
* Improved the ToC and doc structure.
* Added the Meter section to the metrics doc.

### CODE REVIEW

File: docs/logs/README.md [hunk 1]
```diff
@@ -72,6 +73,42 @@ package, regardless of the .NET runtime version being used:
...
+  (e.g. [ASP.NET Core](https://learn.microsoft.com/aspnet/core) and [.NET
+  Worker](https://learn.microsoft.com/dotnet/core/extensions/workers)), in most
+  cases you should use the logger provided by DI, there are special cases when
+  you want log before DI logging pipeline is available or after DI logging
+  pipeline is disposed. Refer to the [.NET official
+  document](https://learn.microsoft.com/dotnet/core/extensions/logging#integration-with-hosts-and-dependency-injection)
+  and [Getting Started with OpenTelemetry .NET Logs in 5 Minutes - ASP.NET Core
+  Application](./getting-started-aspnetcore/README.md) tutorial to learn more.
+* If you are building an application without DI, create a
+  [LoggerFactory](#loggerfactory) instance and configure OpenTelemetry to work
```
Comment (cijothomas) [inline docs/logs/README.md:105]: would be nice to call out to cache the logger instead of repeatedly asking the logger factory

**Review by cijothomas — APPROVED** [review]
LGTM.
Left a comment about explicitly recommending to cache ILogger in non DI apps.

File: docs/logs/README.md (hunk 1 above)
Comment (reyang) [reply docs/logs/README.md:105]: Good callout. Let me add an actual benchmark for this.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: test/Benchmarks/Logs/LogBenchmarks.cs [hunk 2]
```diff
@@ -36,34 +37,39 @@ public class LogBenchmarks
     private readonly ILogger loggerWithTwoProcessors;
     private readonly ILogger loggerWithThreeProcessors;
 
+    private readonly ILoggerFactory loggerFactoryWithNoListener;
```
Comment (utpilla) [inline test/Benchmarks/Logs/LogBenchmarks.cs:40]: We could dispose these logger factories after the benchmarks have run, either in GlobalCleanup or implement IDisposable.

Comment (reyang) [reply test/Benchmarks/Logs/LogBenchmarks.cs:40]: Good call, let me fix it.

Comment (reyang) [reply test/Benchmarks/Logs/LogBenchmarks.cs:40]: #5330

**Pull request merged**